internal/grpc/commander/connection.go

```diff
@@ -117,11 +377,7 @@ func (c *connection) handleAgentConnectRequest(ctx context.Context, cmd *proto.C
 		c.logger.Error(err, "failed to register agent")
 	}
 
-	res := createAgentConnectResponseCmd(
-		cmd.GetMeta().GetMessageId(),
-		requestStatusCode,
-		msg,
-	)
+	res := createAgentConnectResponseCmd(cmd.GetMeta().GetMessageId(), requestStatusCode, msg)
```

Comment:
Is there a possibility that createAgentConnectResponseCmd will be sent after the initialConfig is sent (which is triggered in register() above) ? Can this create a problem?

Reply:
Yes, it is possible, but the agent doesn't care about the order in which it receives those two commands. We just need to make sure we have the `nginxID` and `systemID` set before sending the initial config.